src/controllers/expenses.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+const expenseService = require('../services/expenses');
+const userServices = require('../services/users');
+
+const getAll = (req, res) => {
+  const filterParams = req.query;
+
+  const expenses = expenseService.getAll();
+  const filteredExpenses
+    = expenseService.filterExpenses(expenses, filterParams);
+
+  res.send(filteredExpenses);
+};
+
+const getOne = (req, res) => {
+  const { expenseId } = req.params;
+
+  const expense = expenseService.getOne(+expenseId);
+
+  if (isNaN(+expenseId)) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.send(expense);
+};
+
+const addExpense = (req, res) => {
+  const {
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  } = req.body;
+
+  const user = userServices.getOne(+userId);
+
+  if ((typeof spentAt !== 'string')
+    || (typeof title !== 'string')
+    || (typeof amount !== 'number')
+    || (typeof category !== 'string')
+    || (note && typeof note !== 'string')
+    || !user) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const newExpense = expenseService.addExpense(req.body);
+
+  res.statusCode = 201;
+  res.send(newExpense);
+};
+
+const updateExpense = (req, res) => {
+  const { expenseId } = req.params;
+
+  const expense = expenseService.getOne(+expenseId);
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  const {
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  } = req.body;
+
+  if ((spentAt && typeof spentAt !== 'string')
+    || (title && typeof title !== 'string')
+    || (amount && typeof amount !== 'number')
+    || (category && typeof category !== 'string')
+    || (note && typeof note !== 'string')) {
```

Comment:
same here, let's create separate boolean variable that we'll use inside the `if` statement